### PR TITLE
QRcode, change INTER_LINEAR to INTER_LINEAR_EXACT

### LINKED ISSUE #22133: detectMulti() x86-64 and detectMulti() ARM64 work differently in QRCodeDetector in some cases (problem with resize)
QRCodeDetector.detectMulti() x86-64 and QRCodeDetector.detectMulti() ARM64 work differently in some cases

- OpenCV => 4.6.0 Python
- Operating System / Platform => x86/64 ubuntu 20.04 and jetson nano ARM64

##### Detailed description
script:
```
import cv2 as cv
def main():
    print(cv.__version__)
    image = cv.imread("image056.jpg", cv.IMREAD_IGNORE_ORIENTATION)
    detector = cv.QRCodeDetector()
    res = detector.detectAndDecodeMulti(image)
    print(res)
```
result x86/x64:
```
4.6.0
(False, (), None, ())
```
ARM64 detect and decode successfully:
```
corners: 7.21591736e+02, 2.26398071e+02, 1446., 592., 9.76341980e+02, 1.28422656e+03, 178., 807.
decoded_info: "http://www.bestekmall.com/"
```

image056.jpg:

image056

image from https://boofcv.org/notwiki/regression/fiducial/qrcodes_v3.zip

also some differently values in "detection/nominal/image001.jpg", "detection/nominal/image047.jpg", ...

##### Issue submission checklist

 - [x] I report the issue, it's not a question
 - [x] I checked the problem with documentation, FAQ, open issues,
 - [x] I updated to the latest OpenCV version and the issue is still there
 - [x] There is reproducer code and related data files: videos, images, onnx, etc

### PR DESCRIPTION
**Pull request by AleksandrPanov**
fixes #22133
merge with https://github.com/opencv/opencv_extra/pull/992

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [x] There is a reference to the original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

File: modules/objdetect/misc/python/test/test_qrcode_detect.py [hunk 1]
```diff
@@ -43,10 +43,10 @@ def test_detect_and_decode_multi(self):
...
-        self.assertEqual(decoded_data[2], "SKIP")
-        self.assertEqual(decoded_data[3], "STEP FORWARD")
-        self.assertEqual(decoded_data[4], "STEP BACK")
-        self.assertEqual(decoded_data[5], "QUESTION")
+        self.assertTrue("TWO STEPS FORWARD" in decoded_data)
+        self.assertTrue("EXTRA" in decoded_data)
+        self.assertTrue("SKIP" in decoded_data)
+        self.assertTrue("STEP FORWARD" in decoded_data)
+        self.assertTrue("STEP BACK" in decoded_data)
+        self.assertTrue("QUESTION" in decoded_data)
```
Comment (asmorkalov) [inline modules/objdetect/misc/python/test/test_qrcode_detect.py:51]: Why? The change looks very suspicious.

Comment (AleksandrPanov) [reply modules/objdetect/misc/python/test/test_qrcode_detect.py:51]: this avoids the problem of the order of the found qr codes

**Review by asmorkalov — APPROVED** [review]
:+1: Looks good for me, but fixes in extra part are required.

**Comment by AleksandrPanov** [comment]
@asmorkalov, I opened a [new PR](https://github.com/opencv/opencv/pull/22382) for 3.4. There are differences in the code for 3.4 (another number of resize() call). After merge #22382 to 4.x need to merge #22358.

**Review by asmorkalov — APPROVED** [review]
:+1:

**Pull request merged**